File: test/resources/repos/go/test_repo/child.go
package main

import "fmt"

// ChildStruct embeds BaseStruct, inheriting its fields and methods
type ChildStruct struct {
	BaseStruct // Embedded struct - this is Go's way of inheritance
	Value      int
}

// Process implements the Processable interface
func (c *ChildStruct) Process() error {
	fmt.Printf("Processing child %s with value %d\n", c.Name, c.Value)
	return nil
}

// GetType implements the Processable interface
func (c *ChildStruct) GetType() string {
	return "ChildStruct"
}

// Execute overrides the base Execute method
func (c *ChildStruct) Execute() {
	fmt.Printf("Executing child %s with value %d\n", c.Name, c.Value)
}

// GetValue returns the child's specific value
func (c *ChildStruct) GetValue() int {
	return c.Value
}

File: test/resources/repos/rust/test_repo/src/child.rs
use crate::base::{BaseStruct, Processable, Worker};

// ChildStruct "inherits" from BaseStruct through composition
#[derive(Debug)]
pub struct ChildStruct {
    pub base: BaseStruct,  // Composition - Rust's way of inheritance
    pub value: i32,
}

impl ChildStruct {
    pub fn new(name: String, id: u32, value: i32) -> Self {
        Self {
            base: BaseStruct::new(name, id),
            value,
        }
    }

    pub fn get_value(&self) -> i32 {
        self.value
    }
}

// Implement traits for ChildStruct
impl Processable for ChildStruct {
    fn process(&self) -> Result<(), String> {
        println!("Processing child {} with value {}", self.base.name, self.value);
        Ok(())
    }

    fn get_type(&self) -> String {
        "ChildStruct".to_string()
    }
}

impl Worker for ChildStruct {
    fn execute(&self) {
        println!("Executing child {} with value {}", self.base.name, self.value);
    }

    fn get_worker_id(&self) -> u32 {
        self.base.id
    }
}

// Delegate methods to base
impl std::ops::Deref for ChildStruct {
    type Target = BaseStruct;

    fn deref(&self) -> &Self::Target {
        &self.base
    }
}